src/main/kotlin/org/partiql/scribe/targets/spark/SparkExpandStruct.kt

```diff
@@ -0,0 +1,191 @@
+package org.partiql.scribe.targets.spark
+
+import org.partiql.plan.Fn
+import org.partiql.plan.Rex
+import org.partiql.plan.rex
+import org.partiql.plan.rexOpCallStatic
+import org.partiql.plan.rexOpLit
+import org.partiql.plan.rexOpPathKey
+import org.partiql.plan.rexOpStruct
+import org.partiql.plan.rexOpStructField
+import org.partiql.scribe.asNonNullable
+import org.partiql.types.CollectionType
+import org.partiql.types.StaticType
+import org.partiql.types.StructType
+import org.partiql.types.function.FunctionParameter
+import org.partiql.types.function.FunctionSignature
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.stringValue
+import org.partiql.value.symbolValue
+
+private fun StaticType.expand() = this.metas["EXPAND"] == true
+
+internal fun StaticType.toRexSpark(prefixPath: Rex): Rex {
+    if (!this.expand()) {
+        return prefixPath
+    }
```

Comment:
(self-review) Essentially the same as what was deleted from `SparkRewriter`. Adds this additional check to not expand structs/arrays that do not have any excluded fields.